src/dos/drive_local.cpp

```diff
@@ -130,10 +130,12 @@ bool localDrive::FileOpen(DOS_File **file, char *name, uint32_t flags)
 		return false;
 	}
 
-	// Don't allow opening read-only files in write mode.
+	// Don't allow opening read-only files in write mode,
+	// unless configured otherwise
 	FatAttributeFlags test_attr = {};
 	if (((flags & 0xf) == OPEN_WRITE || (flags & 0xf) == OPEN_READWRITE) &&
-	    (GetFileAttr(name, &test_attr) && test_attr.read_only)) {
+	    (GetFileAttr(name, &test_attr) && test_attr.read_only) &&
+	    !always_open_ro_files) {
```

Comment:
Nitpicky but IMO it's more optimal to put this check first since you you would then not need to run the complex attribute fetching if the var is true.